flycheck.el

```diff
@@ -2930,31 +2956,144 @@ Slots:
      in order to be able to present them to the user.
 
      See `flycheck-related-errors`."
-  buffer checker filename line column message level id group)
+  buffer checker filename coords region message level id group)
+
+; Compute and cache coords if nil
+(advice-add 'flycheck-error-coords :after-until
+            (lambda (err)
+              (let ((coords (flycheck-region-to-coords (flycheck-error-region err)
+                                                       (flycheck-error-buffer err))))
+                (setf (flycheck-error-coords err) coords)
+                coords)))
+
+; Define getters and setters for line and column for compatibility
+(defun flycheck-error-line (err)
+  "Return the line of an errror ERR.
+
+Defined for compatibility."
+  (flycheck-coords-start-line (flycheck-error-coords err)))
+
+(gv-define-simple-setter flycheck-error-line
+                         (lambda (err x)
+                           (setf (flycheck-coords-start-line (flycheck-error-coords err)) x)
+                           (setf (flycheck-coords-end-line   (flycheck-error-coords err)) x)))
+
+(defun flycheck-error-column (err)
+  "Return the column of an errror ERR.
+
+Defined for compatibility."
+  (flycheck-coords-start-col (flycheck-error-coords err)))
+
+(gv-define-simple-setter flycheck-error-column
+                         (lambda (err x)
+                           (setf (flycheck-coords-start-col (flycheck-error-coords err)) x)
+                           (setf (flycheck-coords-end-col   (flycheck-error-coords err)) x)))
```

Comment:
We should update the error region here, right?

Reply:
Will this make everything work with error filters, because the way they work is by modifying errors through `setf`? Is this maybe a drawback to our new practice of precomputing the region; ie. when we change the line and column we need to recompute?

Reply:
Also, should we be exposing an API for modifying `start-line` and `end-line` etc that will also update the region? Right now the only way for them to do this is to modify it within the coords, which won't update the error region. Should there be `flycheck-error-start-line` getters and setters to create the illusion that the `start-line` and others are stored directly on the error?